src/software/backend/simulation/physics_ball.h

```diff
@@ -25,13 +25,17 @@ class PhysicsBall
 
     PhysicsBall() = delete;
 
+    // Delete the copy and assignment operators because copying this class causes
+    // issues with the b2World and how it tracks bodies in the world, because as objects
+    // are copied and destroyed, they will destroy the bodies in the b2World as well
+    PhysicsBall& operator=(const PhysicsBall&) = delete;
+    PhysicsBall(const PhysicsBall&)            = delete;
+
     /**
-     * Removes the corresponding ball body from the Box2D world. If this function
-     * is called on a world that does not contain the ball, nothing happens.
-     *
-     * @param world The Box2D world to remove the ball from
+     * Destroys the PhysicsBall object and removes any corresponding bodies from
+     * the physics world if the ball is part of one
      */
-    void removeFromWorld(std::shared_ptr<b2World> world);
+    ~PhysicsBall();
```

Comment:
How will we deal with removing on collision? Will we incorporate something like how https://www.iforce2d.net/b2dtut/removing-bodies does it. Or is it a case we don't need to worry about as we won't ever remove on collision?

Reply:
Currently there is no plan to remove on collision.

Reply:
Why would you want to remove on collision? Just curious.

Reply:
@garethellis0 It was just a case that I know would cause problems. They're might be others that pertain to us aside from collision, just want to be cautious and aware